src/routes/v2/authenticated/sites.ts

```diff
@@ -188,6 +195,40 @@ export class SitesRouter {
     return res.status(404).json({ message: "Unable to get staging status" })
   }
 
+  getLinkCheckerStatus: RequestHandler<
+    { siteName: string },
+    BrokenLinkErrorDto | ResponseErrorBody,
+    never,
+    never,
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { userWithSiteSessionData } = res.locals
+    const result = await this.repoCheckerService.readRepoErrors(
+      userWithSiteSessionData.siteName
+    )
+    if (result.isOk()) {
+      return res.status(200).json(result.value)
+    }
+    return res.status(400).json({ status: "error" })
+  }
+
+  checkLinks: RequestHandler<
+    { siteName: string },
+    BrokenLinkErrorDto | ResponseErrorBody,
+    never,
+    never,
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { userWithSiteSessionData } = res.locals
+    const result = await this.repoCheckerService.checkRepo(
```

Comment:
I was thinking this will be async? But looks like a sync response here to the user

Reply:
i dont understand this comment, will chat offline

Reply:
understand we are returning a loading state if the lock is present, resolving this comment